### ISSUE TITLE
error overriding method with different access levels in inheritors

### ISSUE THREAD

**Issue by unkarjedy** [lampepfl/dotty#19007]
## Compiler version
3.3.1

## Minimized code
**3 Java files**
```java
public interface RunConfiguration extends Cloneable {
  RunConfiguration clone();
}
```
```java
public abstract class RunConfigurationBase<T> extends UserDataHolderBase implements RunConfiguration {
    @Override
    public RunConfiguration clone() {
        return null;
    }
}
```
```java
import java.util.concurrent.atomic.AtomicReference;

public class UserDataHolderBase extends AtomicReference<String> {
    @Override
    protected Object clone() {
        return null;
    }
}

```

**Main Scala File** 
```scala
class MyRunConfigurationScala extends RunConfigurationBase
```

## Output
Compile the code.
Observe compilation error:
```
.../src/main/scala/MyRunConfigurationScala.scala
error overriding method clone in trait RunConfiguration of type (): RunConfiguration;
  method clone² in class UserDataHolderBase of type (): Object has weaker access privileges; it should be public

where:    clone  is a method in trait RunConfiguration
          clone² is a method in class UserDataHolderBase

class MyRunConfigurationScala extends RunConfigurationBase
```

## Expectation
The code compiles fine.
Note that it compiles with Scala 2 compiler.
Also Java compiler compiles fine a similar code:
`MyRunConfigurationJava.java`
```java
public class MyRunConfigurationJava extends RunConfigurationBase {
}
```

Note that this is a minimised example from the Scala Plugin project.
I tried to extract another module to Scala 3 and got this error.
Java classes are from library (IntelliJ SDK).

**Comment by unkarjedy** [comment]
For now, the workaround is to explicitly override the clone method:
```scala
class MyRunConfigurationScala extends RunConfigurationBase {
  override def clone(): RunConfiguration = super.clone()
}
```

**Comment by EugeneFlesselle** [comment]
I believe this is a duplicate of #18654 and is also fixed by the associated PR #18953
